bindings/wasm/examples/worker.test.js

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import '@vitest/web-worker';
+
+import {WebIO} from '@gltf-transform/core';
+import {expect, suite, test} from 'vitest';
+
+import Module from './built/manifold.js';
+import {readMesh, setupIO} from './gltf-io.js';
+import {examples} from './public/examples.js';
+import ManifoldWorker from './worker?worker';
+
+const io = setupIO(new WebIO());
+
+const wasm = await Module();
+wasm.setup();
+
+function initialized(worker) {
+  return new Promise((resolve) => {
+    worker.onmessage = function(e) {
+      if (e.data == null) {
+        resolve();
+      } else {
+        reject();
+      }
+    };
+  });
+}
+
+let objectURL = null;
+
+async function runExample(name) {
+  const worker = new ManifoldWorker();
+  await initialized(worker);
+  return new Promise((resolve, reject) => {
+    worker.onmessageerror = function({data}) {
+      reject(data);
+    };
+
+    worker.onerror = function(e) {
+      reject(e);
+    };
+
+    worker.onmessage = async function(e) {
+      try {
+        URL.revokeObjectURL(objectURL);
+        objectURL = e.data.objectURL;
+        if (objectURL == null) {
+          reject('no objectURL');
+        }
+        const docIn = await io.read(objectURL);
+        const nodes = docIn.getRoot().listNodes();
+        for (const node of nodes) {
+          const mesh = node.getMesh();
+          if (!mesh) {
+            continue;
+          }
+          const attributes = [];
+          const materials = [];
+          const manifoldMesh = readMesh(mesh, attributes, materials);
```

Comment:
Our JS testing is now giving us dramatically more coverage - we're using the worker directly instead of duplicating its functionality, which means we're also round-tripping every test through write-GLB and read-GLB now.